fpsim/base.py

```diff
@@ -322,17 +321,18 @@ def filter(self, criteria=None, inds=None):
         # Create a new People object with the same properties as the original
         filtered = object.__new__(self.__class__) # Create a new People instance
         BasePeople.__init__(filtered) # Perform essential initialization
-        filtered.__dict__ = {k:v for k,v in self.__dict__.items()} # Copy pointers to the arrays in People
+        filtered.__dict__ = sc.cp(self.__dict__) # Copy pointers to the arrays in People
```

Comment:
I think even simpler would just be `filtered.__dict__ = self.__dict__.copy()`, right? (https://stackoverflow.com/questions/2465921/how-to-copy-a-dictionary-and-only-edit-the-copy) My bad with the original version of this!

Reply:
@cliffckerr I also tried that! but it was slower than using sc.cp/copy.copy -- I can't remember the exact number but could re-rerun and give estimates.

Reply:
Actually it seems that i was wrong, that'll teach me to do with n=3. I think sc.cp may have seemed faster before i introduced other changes :thinking:. In any case,  I ran benchmarks in a more systematic way:

##### Without empowerment:
- With `dict.copy()`: Average runtime (over 20 runs): 6.2778 s
- With `sc.cp()`: Average runtime (over 20 runs): 6.398 s

##### With empowerment:
- With `dict.copy()`: Average runtime (over 20 runs): 6.7289 s
- With `sc.cp()`: Average runtime (over 20 runs): 6.88505 s